dbms/src/Common/TiFlashSecurity.h

```diff
@@ -177,11 +177,13 @@ class TiFlashSecurityConfig : public ConfigObject
         return allowed_common_names.count(cert.commonName()) > 0;
     }
 
-    grpc::SslCredentialsOptions readAndCacheSslCredentialOptions()
+    std::optional<grpc::SslCredentialsOptions> readAndCacheSslCredentialOptions()
     {
         std::unique_lock lock(mu);
+        // if ssl_cerd_options_cached = true, it means the same options has already been returned to grpc-core
+        // don't need to return it again if ssl cert is not actually changed
         if (ssl_cerd_options_cached)
-            return options;
+            return {};
```

Comment:
If the related file paths are not changed, but the file is updated, can we automatically reload the SSL credentials?

Reply:
TiDB doesn't support this kind of reload. It might be acceptable?

Reply:
It should work since we check the `lastModifiedTime` in https://github.com/pingcap/tiflash/blob/3ebb102fba17dce3d990d824a9df93d93f1ab766/dbms/src/Common/TiFlashSecurity.h#L297-L309